http/src/request/guild/member/search_guild_members.rs

```diff
@@ -0,0 +1,159 @@
+
+struct SearchGuildMembersFields {
+    query: Option<String>,
+    limit: Option<u64>,
+}
+
+/// Search the members of a specific guild by a query. Requires the [`GUILD_MEMBERS`] intent.
+///
+/// The upper limit to this request is 100. Discord defaults the limit to 1.
+///
+/// # Examples
+///
+/// Get the first 10 members of guild `100` matching `Wumpus`:
+///
+/// ```rust,no_run
+/// use twilight_http::Client;
+/// use twilight_model::id::GuildId;
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+/// let client = Client::new("my token");
+///
+/// let guild_id = GuildId(100);
+/// let members = client.search_guild_members(guild_id, String::from("Wumpus")).limit(10)?.await?;
+/// # Ok(()) }
+/// ```
+///
+/// # Errors
+///
+/// Returns [`SearchGuildMembersError::LimitInvalid`] if the limit is invalid.
+///
+/// [`GUILD_MEMBERS`]: ../../../../twilight_model/gateway/struct.Intents.html#associatedconstant.GUILD_MEMBERS
+/// [`SearchGuildMembersError::LimitInvalid`]: enum.SearchGuildMembersError.html#variant.LimitInvalid
+pub struct SearchGuildMembers<'a> {
+    fields: SearchGuildMembersFields,
+    fut: Option<Pending<'a, Bytes>>,
+    guild_id: GuildId,
+    http: &'a Client,
+}
+
+impl<'a> SearchGuildMembers<'a> {
+    pub(crate) fn new(http: &'a Client, guild_id: GuildId, query: String) -> Self {
+        Self {
+            fields: SearchGuildMembersFields {
+                query: Some(query),
+                limit: None,
+            },
+            fut: None,
+            guild_id,
+            http,
+        }
+    }
+
+    /// Sets the number of members to retrieve per request.
+    ///
+    /// The limit must be greater than 0 and less than 1000.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`SearchGuildMembersError::LimitInvalid`] if the limit is 0 or
+    /// greater than 1000.
+    ///
+    /// [`SearchGuildMembersError::LimitInvalid`]: enum.GetGuildMembersError.html#variant.LimitInvalid
+    pub fn limit(mut self, limit: u64) -> Result<Self, SearchGuildMembersError> {
+        // Using get_guild_members_limit here as the limits are the same
+        // and this endpoint is not officially documented yet.
+        if !validate::search_guild_members_limit(limit) {
+            return Err(SearchGuildMembersError::LimitInvalid { limit });
+        }
+
+        self.fields.limit.replace(limit);
+
+        Ok(self)
+    }
+
+    fn start(&mut self) -> Result<()> {
+        let query = self.fields.query.take();
+        self.fut
+            .replace(Box::pin(self.http.request_bytes(Request::from(
+                Route::SearchGuildMembers {
+                    guild_id: self.guild_id.0,
+                    limit: self.fields.limit,
+                    query: query.unwrap(), // The user cannot set this to None so it's safe
+                },
+            ))));
+
+        Ok(())
+    }
+}
+
+impl Future for SearchGuildMembers<'_> {
```

Comment:
I may be mistaken, but doesn't `poll_req` do this for us??

Reply:
As I copied this from the get guild members code, I was wondering myself and @vivian clarified that it was for performance reasons. So it makes sense to use that here as well

Reply:
Understandable